### PR TITLE
Disable the failing threadpool tests on Windows. They aren't likely to pass ever

### PR DESCRIPTION
**Pull request by dstoup**
@mwoehlke-kitware has determined that this test failure is more likely a failure of Visual Studio to run the test correctly that it is a failure of the underlying code. Considering the test is never likely to pass in Visual Studio, I propose disabling it there so we can finally see clean dashboards.

### CODE REVIEW

**Comment by dstoup** [comment]
Jenkins test this please

File: vital/util/tests/CMakeLists.txt [hunk 1]
```diff
@@ -12,6 +12,9 @@ kwiver_discover_gtests(vital any_converter      LIBRARIES ${test_libraries})
 kwiver_discover_gtests(vital data_stream_reader LIBRARIES ${test_libraries})
 kwiver_discover_gtests(vital string             LIBRARIES ${test_libraries})
 kwiver_discover_gtests(vital string_editor      LIBRARIES ${test_libraries})
-kwiver_discover_gtests(vital thread_pool        LIBRARIES ${test_libraries})
 kwiver_discover_gtests(vital timer              LIBRARIES ${test_libraries})
 kwiver_discover_gtests(vital token_expander     LIBRARIES ${test_libraries})
+
+if (NOT WIN32)
```
Comment (mwoehlke-kitware) [inline vital/util/tests/CMakeLists.txt]: Please consider adding a note here *why* this is being excluded on Windows.

Comment (dstoup) [reply vital/util/tests/CMakeLists.txt]: Good call, will do .. thanks for chiming in. Didn't think I'd get you for a while.

**Comment by mleotta** [comment]
I don't think it's ever a good idea to just disable tests that indicate a real problem in our code, even if we don't know how to fix it.  However, I can also see the value in ignoring tests like these for normal dashboard builds.  So I propose a slightly different solution.  Let's make a CMake option to disable unresolved test failures.  The dashboards can then ignore the tests that we expect to fail regularly, but more advanced users can turn these tests on to try to fix them.  Or we could even have an a dashboard build in the experimental section that builds these tests too so we don't forget about them.

This also encourages people to write tests that fail because we have not implemented some needed feature.  The dashboards don't have to run those tests (until we expect them pass), but developers can turn them on to work on those problems.

I don't want to forget that this test fails.  I want to fix it or work around it some day.

**Comment by mwoehlke-kitware** [comment]
Re: @mleotta's comment, isn't there a way to mark a CTest as XFAIL? That may be applicable here...

**Comment by dstoup** [comment]
> Re: @mleotta's comment, isn't there a way to mark a CTest as XFAIL? That may be applicable here...

I don't know offhand how to do that with CTest, but it could be applicable for sure. If you have an direct pointers or example of how to do that, forward it along. If not, we can investigate.

 That said, my follow-up comment is a bit more important at the moment. The 'Not Run' tests show up as failures which causes more dashboard havoc when these tests don't explicitly fail. I think it's more critical to figure that issue out and get it corrected.

**Pull request closed without merging**